resources/[system]/spawnmanager/spawnmanager.lua

```diff
@@ -208,18 +208,20 @@ function spawnPlayer(spawnIdx, cb)
     spawnLock = true
 
     Citizen.CreateThread(function()
-        -- if the spawn isn't set, select a random one
-        if not spawnIdx then
-            spawnIdx = GetRandomIntInRange(1, #spawnPoints + 1)
-        end
-
         -- get the spawn from the array
         local spawn
 
-        if type(spawnIdx) == 'table' then
-            spawn = spawnIdx
-        else
-            spawn = spawnPoints[spawnIdx]
+        if not spawnIdx and #spawnPoints > 0 then
+            spawn = spawnPoints[math.random(1, #spawnPoints)]
+        elseif spawnIdx and type(spawnIdx) == 'table' then
```

Comment:
1. How about to calculate `spawnPoints` length only once?
2. Excess `if spawnIdx` check.
3. I would use `GetRandomIntInRange`, as Lua `math.random()` is a crap.